src/util.ts

```diff
@@ -15,3 +17,37 @@ export function formatInterpreterError(error: any) {
         )
         .join("\n\t");
 }
+
+/**
+ * Finds all the test files that match a given list of strings. If the list is empty,
+ * it finds all *.test.brs files.
+ * @param filePatterns A list of file path matches from the command line
+ */
+export async function globMatchFiles(filePatterns: string[]) {
+    let parsedPatterns: string[] = [];
+    filePatterns.forEach((match) => {
+        if (path.parse(match).ext === ".brs") {
+            // If the string is a brs file, match anything that ends with this file name.
+            parsedPatterns.push(`*${match}`);
+        } else {
+            // Do a partial match on any files with this string in their name.
+            parsedPatterns.push(`*${match}*.test.brs`);
+
+            // Do a partial match on any directories with this string in their name.
+            parsedPatterns.push(`*${match}*/**/*.test.brs`);
+        }
+    });
+
+    let testsPattern: string = `${process.cwd()}/**/`;
+    if (parsedPatterns.length === 0) {
+        // If the user didn't specify any pattern, just look for .test.brs files.
+        testsPattern += "*.test.brs";
+    } else if (parsedPatterns.length === 1) {
+        testsPattern += parsedPatterns[0];
+    } else {
+        testsPattern += `{${parsedPatterns.join(",")}}`;
+    }
+
+    // exclude node_modules from the test search
+    return fg.sync([testsPattern, `!${process.cwd()}/node_modules/**/*`]);
```

Comment:
So, previously we used an allow-list model and looked for `${process.cwd()}/{test,tests,source,components}/**/*.test.brs`. That doesn't work quite as well if we let them specify a test pattern, because they could specify a relative path that _includes_ one of those directories, e.g. `roca test/foo/bar.test.brs`.

So I'm thinking a block-list model works better, although this is now definitely a breaking change... Oh well, we're still < 1.0.0, so this is still beta?

Reply:
On second thought, maybe it does work just as well, because we have the `/**/` still?

Reply:
Upon investigation, the allow-list doesn't work (at least with the current logic). If we have `roca test/foo/bar.test.brs` for example we get:

```
${process.cwd()}/{test,tests,components,source}/**/*test/foo/bar.test.brs
```

And since there's only one `test` folder in the path to the file, it doesn't match correctly.